### PR TITLE
[Material] Add ability to set shadow color and selected shadow color for chips and for chip themes

### PR DESCRIPTION
**Pull request by rami-a**
Allow developers to specify the shadow color and selected shadow color for chips, either directly or through the `ChipThemeData`.

### CODE REVIEW

File: packages/flutter/lib/src/material/chip.dart
Comment (willlarche) [inline packages/flutter/lib/src/material/chip.dart]: Get another opinion on whether 'normalShadowColor' would be better.

File: packages/flutter/lib/src/material/chip.dart
Comment (willlarche) [inline packages/flutter/lib/src/material/chip.dart]: cut 'to be used'

Comment (willlarche) [reply packages/flutter/lib/src/material/chip.dart]: And below.

**Review by willlarche — APPROVED** [review]
Get review from Hans.

File: packages/flutter/lib/src/material/chip.dart
Comment (rami-a) [reply packages/flutter/lib/src/material/chip.dart]: Done

File: packages/flutter/lib/src/material/chip.dart
Comment (rami-a) [reply packages/flutter/lib/src/material/chip.dart]: I'm leaning towards just `shadowColor` but will see what @HansMuller thinks.

Comment (HansMuller) [reply packages/flutter/lib/src/material/chip.dart]: In the recent BottomNavigationBar change and in the TabBar theme we used selectedFoo and unselectedFoo but we haven't been consistent about the naming convention. Using shadowColor and selectedShadowColor is OK with me.

File: packages/flutter/lib/src/material/chip_theme.dart
Comment (HansMuller) [inline packages/flutter/lib/src/material/chip_theme.dart]: This comment should either weave in a reference to [selectedShadowColor] or include a see also section.

I imagine that typically developers that set this will also set selectedShadowColor.

File: packages/flutter/lib/src/material/chip_theme.dart
Comment (HansMuller) [inline packages/flutter/lib/src/material/chip_theme.dart]: This comment should either weave in a reference to [shadowColor] or include a see also section.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/chip_theme.dart
Comment (rami-a) [reply packages/flutter/lib/src/material/chip_theme.dart]: Good idea, done

File: packages/flutter/lib/src/material/chip_theme.dart
Comment (rami-a) [reply packages/flutter/lib/src/material/chip_theme.dart]: Done

**Pull request merged**
